src/lib/wallet.js

```diff
@@ -33,75 +31,51 @@ if (IN_FRAME) {
-    const { getMiddleware, initMiddleware } = useMiddleware({ store });
+  store.watch(
+    (state, getters) => getters.activeNetwork,
+    async (network, oldNetwork) => {
+      if (initSdkRunning || isEqual(network, oldNetwork)) {
+        return;
+      }
+      initSdkRunning = true;
 
-    if (IN_FRAME) {
-      const getArrayOfAvailableFrames = () => [
-        window.parent,
-        ...times(window.parent.frames.length, (i) => window.parent.frames[i]),
-      ];
+      try {
+        store.commit('setNodeStatus', NODE_STATUS_CONNECTING);
 
-      const connectedFrames = new Set();
-      executeAndSetInterval(
-        () => getArrayOfAvailableFrames()
-          .filter((frame) => frame !== window)
-          .forEach((target) => {
-            if (connectedFrames.has(target)) return;
-            connectedFrames.add(target);
-            const connection = BrowserWindowMessageConnection({ target });
-            const originalConnect = connection.connect;
-            let intervalId;
-            connection.connect = function connect(onMessage) {
-              originalConnect.call(this, (data, origin, source) => {
-                if (source !== target) return;
-                clearInterval(intervalId);
-                onMessage(data, origin, source);
-              });
-            };
-            store.getters['sdkPlugin/sdk'].addRpcClient(connection);
-            intervalId = executeAndSetInterval(() => {
-              if (!getArrayOfAvailableFrames().includes(target)) {
-                clearInterval(intervalId);
-                return;
-              }
-              store.getters['sdkPlugin/sdk'].shareWalletInfo(connection.sendMessage.bind(connection));
-            }, 3000);
-          }),
-        3000,
-      );
-    }
+        let sdk;
 
-    await Promise.all([
-      store.dispatch('initTippingContractInstances'),
-      getMiddleware(),
-    ]);
-    store.commit('setNodeStatus', NODE_STATUS_CONNECTED);
+        if (isSdkReady.value) {
+          [sdk] = await Promise.all([
+            getSdk(),
+            getMiddleware(),
+          ]);
+          sdk.pool.delete(oldNetwork.name);
+          sdk.addNode(network.name, await createNewNodeInstance(network.url), true);
+        } else {
+          await Promise.all([
+            store.dispatch('sdkPlugin/initialize'),
+            getMiddleware(),
+          ]);
+          sdk = await getSdk();
 
-    store.watch(
-      (state, getters) => getters.activeNetwork,
-      async (network, oldNetwork) => {
-        if (isEqual(network, oldNetwork)) return;
-        try {
-          store.commit('setNodeStatus', NODE_STATUS_CONNECTING);
-          await initMiddleware();
-          store.commit('setNodeStatus', NODE_STATUS_CONNECTED);
-        } catch (error) {
-          store.commit('setNodeStatus', NODE_STATUS_ERROR);
-          Logger.write(error);
+          if (IN_FRAME) {
+            FramesConnection.init(sdk);
```

Comment:
Do we need to run this every time we change a network?

Reply:
Good point. But this code is running in the `else` block which is fired only if there's no SDK, which means this is an initial state of the app.
To be 100% sure I added `initialized` flag to the lib. WHat do you think about this?